internal/oscal/generate_test.go

```diff
@@ -29,3 +60,171 @@ func TestFmtFieldName(t *testing.T) {
 		}
 	}
 }
+
+// TestGenerateUniqueIdMapId tests that the generateUniqueIdMap function
+// returns the correct '$id'.
+func TestGenerateUniqueIdMapId(t *testing.T) {
+	oscalMap, err := parseOscalFileToMap()
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, actualId := generateUniqueIdMap(oscalMap)
+
+	expectedId := "#assembly_oscal-component-definition_component-definition"
+
+	if expectedId != actualId {
+		t.Errorf("error generateUniqueIdMap(): expected: %s | got: %s", expectedId, actualId)
+	}
+}
+
+// TestGenerateUniqueIdMapProperties tests that the generateUniqueIdMap function
+// returns the correct 'properties'.
+func TestGenerateUniqueIdMapProperties(t *testing.T) {
+	var actualProperties string
+	expectedPropertiesFile := "../../testdata/expected-properties.txt"
+
+	oscalMap, err := parseOscalFileToMap()
+	if err != nil {
+		t.Error(err)
+	}
+
+	actualMap, actualId := generateUniqueIdMap(oscalMap)
+
+	// Check if there's a properties field.
+	// If there is, loop over to collect the properties and assert we have the correct properties.
+	// If there isn't, fail the test and print error message.
+	if properties := actualMap[actualId].(map[string]interface{})["properties"]; properties != nil {
+		properties := actualMap[actualId].(map[string]interface{})["properties"]
+
+		// Store the properties fields to a string slice and sort it.
+		propertiesSlice := make([]string, 0)
+		for property := range properties.(map[string]interface{}) {
+			propertiesSlice = append(propertiesSlice, property)
+		}
+		sort.Strings(propertiesSlice)
+
+		// Convert the sorted string slice to a string so that we can assert the output.
+		for _, property := range propertiesSlice {
+			actualProperties += fmt.Sprintf("%s\n", property)
+		}
+	} else {
+		t.Error("The 'properties' field was not found. Please verify the OSCAL JSON schema file is valid.")
+	}
+
+	expectedPropertiesBytes, err := os.ReadFile(expectedPropertiesFile)
+	if err != nil {
+		t.Error(err)
+	}
+
+	expected := string(expectedPropertiesBytes)
+
+	actual := strings.TrimSpace(actualProperties)
+
+	if expected != actual {
+		t.Errorf("error generateUniqueIdMap():\n\nexpected: \n%s\n\ngot: \n%s", expected, actual)
+	}
+}
+
+// TestFormatStructTags tests that we can format Go struct tags correctly.
+func TestFormatStructTags(t *testing.T) {
+	oscalMap, err := parseOscalFileToMap()
+	if err != nil {
+		t.Error(err)
+	}
+
+	idMap, id := generateUniqueIdMap(oscalMap)
+
+	actualTagList := formatStructTags(idMap, id, "uuid", []string{"json", "yaml"})
```

Comment:
[no change required] - `formatStructTags()` use of `idMap` and `id` almost look completely transient. We should consider calling `getRequiredFields()` prior to invoking it.

Reply:
Yeah they are pretty much just acting as placeholders in the function parameters. What is the ideal outcome in this test? Are you wanting to test adding `omitempty` to the tags?

Reply:
I think there's going to be more refactoring involved than simply calling the `getRequiredFields()`  function prior to calling `formatStructTags()`  since they are pretty tightly coupled at the moment. Good to make note of for future iterations though